packages/components/src/Popover/Popover.test.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useRef } from "react";
+import {
+  RenderResult,
+  act,
+  cleanup,
+  fireEvent,
+  render,
+} from "@testing-library/react";
+import { Popover } from ".";
+import { PopoverProps } from "./Popover";
+
+afterEach(cleanup);
+
+let rendered: RenderResult;
+const content = "Test Content";
+
+const PopoverTestComponent = (props: PopoverProps) => {
+  const checkboxRef = useRef();
+  return (
+    <>
+      <div ref={checkboxRef}></div>
+      <Popover {...props} attachTo={checkboxRef}>
+        {props.children}
+      </Popover>
+    </>
+  );
+};
+
+it("should render a Popover with the content and dismiss button", async () => {
+  const handleClose = jest.fn();
+
+  await act(async () => {
+    rendered = render(
+      <PopoverTestComponent
+        attachTo={undefined}
+        open={true}
+        onRequestClose={handleClose}
+      >
+        {content}
+      </PopoverTestComponent>,
+    );
+  });
+
+  const { queryByText, queryByTestId, container } = rendered;
+
+  expect(queryByText(content)).toBeTruthy();
+  expect(queryByTestId("popover")).not.toBeNull();
+
+  fireEvent.click(container.querySelector("[aria-label='Close modal']"));
```

Comment:
Would be nicer to query for the text `Close Modal` rather than doing a css selector here.

Reply:
But it's a dismiss button with an `X` icon so can't query by text unfortunately.

Reply:
If you query by label it should find the aria label as well.